src/pb_stub.cc

```diff
@@ -1346,8 +1352,39 @@ Logger::BackendLoggingActive()
 
 PYBIND11_EMBEDDED_MODULE(c_python_backend_utils, module)
 {
+  py::enum_<TRITONSERVER_Error_Code>(module, "ErrorCode")
```

Comment:
Is it possible the make the `enum` part of the `TritonError` class?

Reply:
Sure. It is updated to:
```
        error_code_map = {
            "UNKNOWN": pb_utils.TritonError.ErrorCode.UNKNOWN,
            "INTERNAL": pb_utils.TritonError.ErrorCode.INTERNAL,
            "NOT_FOUND": pb_utils.TritonError.ErrorCode.NOT_FOUND,
            "INVALID_ARG": pb_utils.TritonError.ErrorCode.INVALID_ARG,
            "UNAVAILABLE": pb_utils.TritonError.ErrorCode.UNAVAILABLE,
            "UNSUPPORTED": pb_utils.TritonError.ErrorCode.UNSUPPORTED,
            "ALREADY_EXISTS": pb_utils.TritonError.ErrorCode.ALREADY_EXISTS,
        }
```

Reply:
Maybe just Code instead of ErrorCode?

Ex: `TritonError.Code.INTERNAL` instead `TritonError.ErrorCode.INTERNAL`? Similar to `Status::Code::INTERNAL`, and also matches python interface `pb_utils.TritonError(code=...)`.

Reply:
I'd say to remove the `ErrorCode` completely if possible. For example,

```
error_code_map = {
            "UNKNOWN": pb_utils.TritonError.UNKNOWN,
            "INTERNAL": pb_utils.TritonError.INTERNAL,
            "NOT_FOUND": pb_utils.TritonError.NOT_FOUND,
            "INVALID_ARG": pb_utils.TritonError.INVALID_ARG,
            "UNAVAILABLE": pb_utils.TritonError.UNAVAILABLE,
            "UNSUPPORTED": pb_utils.TritonError.UNSUPPORTED,
            "ALREADY_EXISTS": pb_utils.TritonError.ALREADY_EXISTS,
        }
```

Reply:
Yes, it is updated: [Expose ErrorCode internal in TritonError](https://github.com/triton-inference-server/python_backend/pull/292/commits/0a7a8b96f36e6302d5f60790a20a85e08ce5595e)

It looks like it is impossible to define readonly static objects inside `TritonError` without the `__ErrorCode` enum defined somewhere. My solution above is to hide the `__ErrorCode` inside `TritonError`, so the actual error codes can be defined directly within the `TritonError` object. I wonder if you might know a cleaner solution to the one above?